lib/gemsurance/gem_info_retriever.rb

```diff
@@ -148,5 +148,13 @@ def retrieve(options = {})
       end
       gem_infos
     end
+
+    private
+
+    def get_gem_info(gem_name)
+      Gems.info(gem_name)
+    rescue StandardError
+      {}
```

Comment:
OK, I did a little digging here. I suppose the idea is to guard against any network errors here? 
https://github.com/rubygems/gems/blob/v0.8.0/lib/gems/request.rb
The newer versions will have a check for JSON parsing errors here:
https://github.com/rubygems/gems/pull/41
I don't know which errors you are experiencing here?
In any case, could you output a warning here?
Something like:
```
rescue StandardError => e
  $stderr.puts e
  {}
end
```